src/Arcus.Observability.Tests.Unit/Serilog/CloudContextConverterTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Arcus.Observability.Telemetry.Core;
+using Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights.Converters;
+using Bogus;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Moq;
+using Serilog.Events;
+using Serilog.Parsing;
+using Xunit;
+
+namespace Arcus.Observability.Tests.Unit.Serilog
+{
+    [Trait("Category", "Unit")]
+    public class CloudContextConverterTests
+    {
+        private readonly Faker _bogusGenerator = new Faker();
+
+        [Fact]
+        public void EnrichWithAppInfo_TraceWithComponentNameAndPodName_CloudContextRoleNameAndRoleInstanceSet()
+        {
+            // Arrange
+            string roleName = _bogusGenerator.Name.JobArea();
+            string roleInstance = _bogusGenerator.Name.JobType() + _bogusGenerator.Random.Number();
+
+            LogEvent logEvent = CreateLogEvent(
+                new LogEventProperty(ContextProperties.General.ComponentName, new ScalarValue(roleName)),
+                new LogEventProperty(ContextProperties.Kubernetes.PodName, new ScalarValue(roleInstance)));
+            
+            var telemetry = new TraceTelemetry();
+            var converter = new CloudContextConverter();
+
+            // Act
+            converter.EnrichWithAppInfo(logEvent, telemetry);
+
+            // Assert
+            Assert.Equal(roleName, telemetry.Context.Cloud.RoleName);
+            Assert.Equal(roleInstance, telemetry.Context.Cloud.RoleInstance);
+        }
+
+        [Fact]
+        public void EnrichWithAppInfo_EventWithComponentNameAndMachineName_CloudContextRoleNameAndRoleInstanceSet()
```

Comment:
I would add one more where PodName & MachineName are present to assert that the correct one wins

Reply:
Ah, yes, sorry about that. Shame I can't use property-based testing thoug 😄 .